One NPE exception happened in FiberSensor.scala when running tpcds

During running tpcds queries with the laste oap master code, below one null pointer exception are found.

The NPE happens at below line of FiberSensor.scala.
      private def conf = OapRuntime.getOrCreate.sparkSession.conf

Below is the detailed exception stack.
Exception in thread "broadcast-exchange-0" java.lang.ExceptionInInitializerError
	at org.apache.spark.sql.execution.datasources.oap.filecache.FiberSensor.getHosts(FiberSensor.scala:116)
	at org.apache.spark.sql.execution.FileSourceScanExec$$anonfun$28$$anonfun$apply$5.apply(DataSourceScanExec.scala:452)
	at org.apache.spark.sql.execution.FileSourceScanExec$$anonfun$28$$anonfun$apply$5.apply(DataSourceScanExec.scala:440)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at org.apache.spark.sql.execution.FileSourceScanExec$$anonfun$28.apply(DataSourceScanExec.scala:440)
	at org.apache.spark.sql.execution.FileSourceScanExec$$anonfun$28.apply(DataSourceScanExec.scala:439)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.immutable.List.flatMap(List.scala:344)
	at org.apache.spark.sql.execution.FileSourceScanExec.createNonBucketedReadRDD(DataSourceScanExec.scala:439)
	at org.apache.spark.sql.execution.FileSourceScanExec.inputRDD$lzycompute(DataSourceScanExec.scala:308)
	at org.apache.spark.sql.execution.FileSourceScanExec.inputRDD(DataSourceScanExec.scala:286)
	at org.apache.spark.sql.execution.FileSourceScanExec.inputRDDs(DataSourceScanExec.scala:313)
	at org.apache.spark.sql.execution.FilterExec.inputRDDs(basicPhysicalOperators.scala:124)
	at org.apache.spark.sql.execution.ProjectExec.inputRDDs(basicPhysicalOperators.scala:42)
	at org.apache.spark.sql.execution.WholeStageCodegenExec.doExecute(WholeStageCodegenExec.scala:386)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$execute$1.apply(SparkPlan.scala:117)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$execute$1.apply(SparkPlan.scala:117)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$executeQuery$1.apply(SparkPlan.scala:138)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:135)
	at org.apache.spark.sql.execution.SparkPlan.execute(SparkPlan.scala:116)
	at org.apache.spark.sql.execution.SparkPlan.getByteArrayRdd(SparkPlan.scala:228)
	at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:275)
	at org.apache.spark.sql.execution.exchange.BroadcastExchangeExec$$anonfun$relationFuture$1$$anonfun$apply$1.apply(BroadcastExchangeExec.scala:76)
	at org.apache.spark.sql.execution.exchange.BroadcastExchangeExec$$anonfun$relationFuture$1$$anonfun$apply$1.apply(BroadcastExchangeExec.scala:73)
	at org.apache.spark.sql.execution.SQLExecution$.withExecutionId(SQLExecution.scala:103)
	at org.apache.spark.sql.execution.exchange.BroadcastExchangeExec$$anonfun$relationFuture$1.apply(BroadcastExchangeExec.scala:72)
	at org.apache.spark.sql.execution.exchange.BroadcastExchangeExec$$anonfun$relationFuture$1.apply(BroadcastExchangeExec.scala:72)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
	at org.apache.spark.sql.execution.datasources.oap.filecache.FiberSensor$.conf(FiberSensor.scala:129)
	at org.apache.spark.sql.execution.datasources.oap.filecache.FiberSensor$.<init>(FiberSensor.scala:131)
	at org.apache.spark.sql.execution.datasources.oap.filecache.FiberSensor$.<clinit>(FiberSensor.scala)

--- a/src/main/spark2.1/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala
+++ b/src/main/spark2.1/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala
@@ -68,7 +68,7 @@ private[spark] object OapEnv extends Logging {
 
       SparkSQLEnv.sparkContext = sparkContext
       SparkSQLEnv.sqlContext = sqlContext
-      this.sparkSession = sparkSession
+      this.sparkSession = sqlContext.sparkSession
 
       sparkContext.ui.foreach(new OapTab(_))
       initialized = true

--- a/src/main/spark2.2/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala
+++ b/src/main/spark2.2/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala
@@ -71,7 +71,7 @@ private[spark] object OapEnv extends Logging {
 
       SparkSQLEnv.sparkContext = sparkContext
       SparkSQLEnv.sqlContext = sqlContext
-      this.sparkSession = sparkSession
+      this.sparkSession = sqlContext.sparkSession
 
       sparkContext.ui.foreach(new OapTab(_))
       initialized = true

--- a/src/main/spark2.3/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala
+++ b/src/main/spark2.3/scala/org/apache/spark/sql/hive/thriftserver/OapEnv.scala
@@ -71,7 +71,7 @@ private[spark] object OapEnv extends Logging {
 
       SparkSQLEnv.sparkContext = sparkContext
       SparkSQLEnv.sqlContext = sqlContext
-      this.sparkSession = sparkSession
+      this.sparkSession = sqlContext.sparkSession
 
       sparkContext.ui.foreach(new OapTab(_))
       initialized = true

